cluster-autoscaler/simulator/cluster.go

```diff
@@ -113,14 +112,25 @@ type UtilizationInfo struct {
 
 // FindNodesToRemove finds nodes that can be removed. Returns also an information about good
 // rescheduling location for each of the pods.
-func FindNodesToRemove(candidates []*apiv1.Node, destinationNodes []*apiv1.Node, pods []*apiv1.Pod,
-	listers kube_util.ListerRegistry, clusterSnapshot ClusterSnapshot, predicateChecker PredicateChecker, maxCount int,
-	fastCheck bool, oldHints map[string]string, usageTracker *UsageTracker,
+func FindNodesToRemove(
+	candidates []*schedulernodeinfo.NodeInfo,
+	destinationNodes []*schedulernodeinfo.NodeInfo,
```

Comment:
I think we already had this discussion, but I think passing along list of NodeInfos together with clusterSnapshot everywhere is really ugly. 
Snapshot should be the source of truth for NI. But we end up using both the snapshot and the list passed as parameter. I'm not saying building NI map at every step (ie. current state) is any better, but I think it would be more consistent to add getters based on node name to snapshot (GetNode(nodeName), GetPodsForNode(nodeName)) and just pass list of node names and snapshot. WDYT?
cc: @losipiuk for another opinion on this

Reply:
We already have 'Get(nodeName)', which returns nodeInfo, which includes pods ;) 

I didn't want to bundle too many changes, but if you think it's worth refactoring the node list processors to return names, I'm happy to do this.

Reply:
I added the changes in new commits. The biggest drawback is IMO redundant error-checking, but it's not easy to get rid of since Get() can fail if a node is not in the snapshot.

Reply:
I generally agree with Maciek's comment here. I would prefer to limit use of `NodeInfo` in the codebese. Preferably to 0. As this is internal scheduler structure.